app/controllers/application_controller.rb

```diff
@@ -1014,17 +1014,28 @@ def query_params(query = nil)
   helper_method :query_params
 
   def add_query_param(params, query = nil)
+    if !browser.bot?
+      params[:q] = get_query_param(query)
+    end
+    params
+  end
+  helper_method :add_query_param
+
+  # Allows us to add query to a path helper:
+  #   object_path(@object, q: get_query_param)
+  def get_query_param(query = nil)
```

Comment:
`ApplicationController#get_query_param` is a bit of @nimmolo's magic from `bootstrap-4`. 
He has some other magic there which probably is worth porting -- e.g., methods to literally use `object_path(@object, q: get_query_param)`. 
I'm not sure where all of them are.
I also would be more comfortable if there were a reasonable way to test  ApplicationController.

Reply:
Those were added in `helpers/object_link_helpers`.
See below for reference to the `routeable` module.

I feel like testing the methods that depend on methods in application_controller might be coverage enough...